Add profiler installation

labbati opened the pull request:
### Description

Add profiling module installation to the new php installation script.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

morrisonlevi reviewed (commented):
When installing on the 7.1-buster container, I got this:

```
Cannot load datadog-profiling - it was built with configuration API320160303,NTS, whereas running engine is API320160303,NTS,debug
Cannot load datadog-profiling - it was built with configuration API320160303,NTS, whereas running engine is API320160303,NTS,debug
Cannot load datadog-profiling - it was built with configuration API320160303,NTS, whereas running engine is API320160303,NTS,debug
```

Glancing through the changes, I don't see code for debug/zts incompatibilities, so this makes sense, I guess. I thought I saw code for this in a prior glance through at this PR; was it accidentally removed?

After switching to the nts, ndebug variant it installed cleanly and CLI diagnostics succeeded:

```
DD_AGENT_HOST=agent DD_PROFILING_ENABLED=1 php --ri=datadog-profiling

datadog-profiling

                      Profiling Recorder Diagnostics                      
Enabled => yes
Can create profiles => yes
                       Profiling Upload Diagnostics                       
info => [Datadog Profiling] Successfully uploaded profile.

Can upload profiles => yes
```

And I correctly have an empty profile in my Datadog UI from running the phpinfo diagnostic.

I think I probably need to publish rc.5 which disables the profiler by default, don't I? To make the beta -> GA transition smoother (people won't need to adjust variables?).

labbati commented:
Very good catch @morrisonlevi. It also uncovered a bug (zts) and an unhandled scenario (debug-zts). Added tests for all of those and fixed them.

I manually tested them replaing the github.com url with the circleci url locally.

morrisonlevi commented on Makefile:
```diff
@@ -293,30 +293,11 @@ $(PACKAGES_BUILD_DIR):
+	bash ./tooling/bin/generate-final-artifact.sh \
+		$(VERSION) \
+		$(PACKAGES_BUILD_DIR) \
+		https://github.com/DataDog/dd-prof-php/releases/download/v0.3.0-rc.4/datadog-profiling.tar.gz
```
```suggestion
		https://github.com/DataDog/dd-prof-php/releases/download/v0.3.0-rc.5/datadog-profiling.tar.gz
```

labbati replied:
I will make this change in the release branch (and add it to the release steps)

morrisonlevi reviewed (commented):
Not a blocker for this PR unless you want it to be:

When a user has a build of PHP that isn't compatible with profiling (debug build, or unsupported version like PHP 5), I don't think we should silently proceed without the profiler. Since enabling profiling is opt-in, not opt-out, then if they say `--enable-profiling` and it's on a version or build of PHP that doesn't support it, then "succeeding" but actually not installing the profiler doesn't seem right to me. That's my $0.02.

labbati commented:
> then "succeeding" but actually not installing the profiler doesn't seem right to me. That's my $0.02.
I agree, taking a note and will change immediately after the release
